drake/examples/Pendulum/Pendulum.h

```diff
@@ -3,100 +3,137 @@
-    Eigen::MatrixXd R(1,1);  R << 1;
-    Eigen::VectorXd xG(2);   xG << M_PI, 0;
-    Eigen::VectorXd uG(1);   uG << 0;
 
-    return timeInvariantLQR(xG,uG,Q,R);
-  }
+template <typename ScalarType = double>
+class PendulumInput {
+public:
+  PendulumInput(void) : tau(0) {};
+  PendulumInput(const Eigen::Matrix<ScalarType,1,1>& x) : tau(x(0)) {};
 
+  PendulumInput& operator=(const Eigen::Matrix<ScalarType,1,1>& x) {
+    tau = x(0);
+    return *this;
+  }
 
-  double m,l,b,lc,I,g;  // pendulum parameters (initialized in the constructor)
+  operator Eigen::Matrix<ScalarType,1,1> () const {
+    Eigen::Matrix<ScalarType,1,1> x;
+    x << tau;
+    return x;
+  }
 
-private:
-  template <typename Scalar,typename Vector>
-  Vector dynamics_implementation(Scalar t, const Vector& x, const Vector& u) const {
-    Vector xdot(2);
-    xdot(0) = x(1);
-    xdot(1) = (u(0) - m*g*lc*sin(x(0)) - b*x(1))/I;
-    return xdot;
+  friend std::ostream& operator<<(std::ostream& os, const PendulumInput& x)
+  {
+    os << "  tau = " << x.tau << endl;
+    return os;
   }
 
+  static unsigned int size() { return 1; }
+
+  ScalarType tau;
 };
 
-class PendulumWithBotVis : public Pendulum {
+namespace Drake {
+
+  template <typename ScalarType>
+  struct VectorTraits<PendulumState<ScalarType> > {
+    enum {
+      RowsAtCompileTime = 2
+    };
+  };
+
+  template <typename ScalarType>
+  struct VectorTraits<PendulumInput<ScalarType> > {
+    enum {
+      RowsAtCompileTime = 1
+    };
+  };
+
+}
+
+class Pendulum : public Drake::System<Pendulum,PendulumState,PendulumInput,PendulumState,false,false> {
 public:
-  PendulumWithBotVis(const std::shared_ptr<lcm::LCM>& lcm) : Pendulum(lcm), botvis(lcm,"Pendulum.urdf",DrakeJoint::FIXED) {}
+  Pendulum() :
+          m(1.0), // kg
+          l(.5),  // m
+          b(0.1), // kg m^2 /s
+          lc(.5), // m
+          I(.25), // m*l^2; % kg*m^2
+          g(9.81) // m/s^2
+  {}
+  virtual ~Pendulum(void) {};
 
-  virtual Eigen::VectorXd output(double t, const Eigen::VectorXd& x, const Eigen::VectorXd& u) const override {
-    botvis.output(t,Eigen::VectorXd::Zero(0),x);
-    return Pendulum::output(t,x,u);
+  template <typename ScalarType>
+  PendulumState<ScalarType> dynamicsImplementation(const PendulumState<ScalarType>& x, const PendulumInput<ScalarType>& u) const {
+    PendulumState<ScalarType> dot;
+    dot.theta = x.thetadot;
+    dot.thetadot = (u.tau - m*g*lc*sin(x.theta) - b*x.thetadot)/I;
+    return dot;
   }
 
-  BotVisualizer botvis;
+
+  template <typename ScalarType>
+  PendulumState<ScalarType> outputImplementation(const PendulumState<ScalarType>& x) const {
+    return x;
+  }
+
+public:
+  double m,l,b,lc,I,g;  // pendulum parameters (initialized in the constructor)
 };
 
-class PendulumEnergyShaping : public DrakeSystem {
+
+class PendulumEnergyShaping : public Drake::System<PendulumEnergyShaping,Drake::UnusedVector,PendulumState,PendulumInput,false,true> {
```

Comment:
I know it's been called PendulumEnergyShaping on the Matlab side forever, but I still think that PendulumEnergyShapingController would be clearer.